src/Microsoft.Management.Configuration/Microsoft.Management.Configuration.idl

```diff
@@ -0,0 +1,605 @@
+    {
+        Unknown,
+        // Indicates that the first configuration set has a matching name and origin to the second, which has already been applied.
+        // This is likely an update to the existing set, and should be applied as such, rather than an entirely new set.
+        MatchingOrigin,
+        // Indicates that the first configuration set is identical to the second, which has already been applied.
+        // This is based solely on the configuration unit settings.
+        IdenticalSetApplied,
+        // Indicates a conflict between the settings of two configuration units.
+        SettingsConflict,
+    };
+
+    // Describes a conflict between a setting of two configuration units.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    runtimeclass ConfigurationConflictSetting
+    {
+        // The name of the setting.
+        String Name{ get; };
+
+        // The value from the first configuration unit.
+        // These are the values from a `ValueSet`, and are thus required to be a `PropertyValue` or a 'ValueSet`.
+        Object FirstValue{ get; };
+
+        // The value from the second configuration unit.
+        // These are the values from a `ValueSet`, and are thus required to be a `PropertyValue` or a 'ValueSet`.
+        Object SecondValue{ get; };
+    }
+
+    // Describes a conflict between two configuration sets.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    runtimeclass ConfigurationConflict
+    {
+        // The type of conflict detected.
+        ConfigurationConflictType Conflict{ get; };
+
+        // The first of the configuration sets involved in the conflict.
+        ConfigurationSet FirstSet{ get; };
+
+        // The second of the configuration sets involved in the conflict.
+        ConfigurationSet SecondSet{ get; };
+
+        // The first of the configuration units involved in the conflict.
+        ConfigurationUnit FirstUnit{ get; };
+
+        // The second of the configuration units involved in the conflict.
+        ConfigurationUnit SecondUnit{ get; };
+
+        // Contains information about the particular settings that are conflicting.
+        Windows.Foundation.Collections.IVectorView<ConfigurationConflictSetting> Settings{ get; };
+    }
+
+    // Flags to control how a configuration set should be applied to the system.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    enum ApplyConfigurationSetFlags
+    {
+        None = 0x0,
+        // Forces a new configuration set instance to be recorded when the set being applied matches a previous set's origin.
+        // The default behavior is to assume that the incoming set is an update to the existing set and overwrite it.
+        DoNotOverwriteMatchingOriginSet = 0x1,
+    };
+
+    // The result of applying the settings for a configuration unit.
+    [contract(Microsoft.Management.Configuration.Contract, 1)]
+    runtimeclass ApplyConfigurationUnitResult

```

Comment:
What's the difference with ApplySettingsResult?

Reply:
`ApplyConfigurationUnitResult` is the interface for users of `Microsoft.Management.Configuration`.  `ApplySettingsResult` is the interface for implementors of `IConfigurationUnitProcessor`.